src/auction.js

```diff
@@ -363,9 +363,12 @@ export function newAuction({adUnits, adUnitCodes, callback, cbTimeout, labels, a
     }
   }
 
-  function addWinningBid(winningBid) {
+  function addWinningBid(winningBid, aUnits = adUnits, am = adapterManager) {
```

Comment:
I think this is overdoing dependency injection - `adUnits` is already something you can pass in directly to the constructor; `adapterManger` could also be if you want it injected, it doesn't make a lot of sense to do it in just one method - the next guy will need to reason about an `auction` with tightly coupled methods but completely disjointed depedendencies.